editor/src/components/canvas/controls/selection-area-helpers.ts

```diff
@@ -0,0 +1,150 @@
+  const element = MetadataUtils.findElementByElementPath(metadata, path)
+  if (element == null || element.globalFrame == null) {
+    return null
+  }
+  if (!isFiniteRectangle(element.globalFrame)) {
+    return null
+  }
+  return element.globalFrame
+}
+
+export const filterUnderSelectionArea = (
+  paths: ElementPath[],
+  metadata: ElementInstanceMetadataMap,
+  area: CanvasRectangle | null,
+  selectedViews: ElementPath[],
+): ElementPath[] => {
+  if (area == null) {
+    return []
+  }
+
+  const elements: ElementUnderSelectionArea[] = paths.map((path) => {
+    const frame = maybeElementFrame(path, metadata)
+    return {
+      path: path,
+      type: getElementUnderSelectionAreaType(metadata, path),
+      fullyContained: frame != null && rectangleContainsRectangle(area, frame),
+      selected: EP.containsPath(path, selectedViews),
+      zeroSized: frame == null || frame.width === 0 || frame.height === 0,
+    }
+  })
+
+  const isSceneChild =
+    (element: ElementUnderSelectionArea) => (other: ElementUnderSelectionArea) => {
+      return other.type === 'scene' && EP.isDescendantOf(element.path, other.path)
+    }
+
+  const thereAreRegularElements = elements.some(
+    (element) => element.type === 'regular' && !elements.some(isSceneChild(element)),
+  )
+
+  return elements
+    .filter((element) => {
+      // no zero-sized elements
+      if (element.zeroSized) {
+        return false
+      }
+
+      // only outermost children
+      if (
+        element.type === 'regular' &&
+        elements.some((other) => {
+          return (
+            other.type !== 'scene' &&
+            !other.zeroSized &&
+            EP.isDescendantOf(element.path, other.path)
+          )
+        })
+      ) {
+        return false
+      }
+
+      const parentScene = elements.find(isSceneChild(element))
+
+      // if the element is a scene child and there are storyboard children, skip it
+      if (parentScene != null && thereAreRegularElements) {
+        return false
+      }
+      // if the element is a scene, and the scene is not fully contained skip the scene
+      if (element.type === 'scene' && !element.fullyContained && !element.selected) {
+        return false
+      }
+      // if a scene is fully contained, select just the scene and omit its children
+      if (parentScene != null && parentScene.fullyContained) {
+        return false
+      }
+      return true
+    })
+    .map((r) => r.path)
+}
+
+export function getSelectionAreaRenderedRect(
+  selectionArea: WindowRectangle | null,
+  boundingRect: DOMRect | null,
+): WindowRectangle | null {
+  if (selectionArea == null || boundingRect == null) {
+    return null
+  }
+  return {
+    x: selectionArea.x - boundingRect.x,
+    y: selectionArea.y - boundingRect.y,
+    width: selectionArea.width,
+    height: selectionArea.height,
+  } as WindowRectangle
+}
+
+export function makeSelectionArea(from: WindowPoint, to: WindowPoint): WindowRectangle {
```

Comment:
This seems to be the same function as `rectFromTwoPoints`

Reply:
it does yes! gonna get rid of it

Reply:
https://github.com/concrete-utopia/utopia/pull/3750/commits/3c01789db22df57059c21634eb41aaf8101f985c